adapter/statsclient/statsclient.go

```diff
@@ -144,24 +152,23 @@ func (sc *StatsClient) Connect() (err error) {
 // Disconnect from the socket, unmap shared memory and terminate
 // socket monitor
 func (sc *StatsClient) Disconnect() error {
-	if sc.headerData == nil {
-		return nil
+	if atomic.CompareAndSwapUint32(&sc.monitored, 1, 0) {
+		close(sc.done)
 	}
-	if err := syscall.Munmap(sc.headerData); err != nil {
```

Comment:
Why remove the unmap?

Reply:
To avoid code duplication. Now only `disconnect()` handles that, and `Disconnect()` calls it when necessary.